app/static/app/js/components/Paginated.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import update from 'react-addons-update';
+
+/*abstract*/ class Paginated extends React.Component{
+  constructor(){
+    super();
+  }
+
+  updatePagination(itemsPerPage, totalItems){
+    let currentPage = 1;
+    const totalPages = this.totalPages(itemsPerPage, totalItems);
+
+    if (this.state.pagination && this.state.pagination.currentPage !== undefined){
```

Comment:
I typically use `!= null` (not `!==`) because it covers both the `null` and `undefined` cases.